src/fg_apply.rs

```diff
@@ -160,11 +160,8 @@ pub(crate) unsafe fn rav1d_apply_grain_row<BD: BitDepth>(
     if data.num_y_points != 0 {
         let bh = cmp::min(h - row * BLOCK_SIZE, BLOCK_SIZE);
         dsp.fgy_32x32xn.call(
-            out_data[0]
-                .as_strided_mut_ptr::<BD>()
-                .offset((row * BLOCK_SIZE) as isize * BD::pxstride(out.stride[0])),
-            luma_src,
```

Comment:
We're now computing `luma_src` in two places, how will we ensure they stay in sync?

Reply:
I don't really buy that as a good practice, but this is ok for now.